### PR TITLE
refactor(validator): moved validator groovy unit tests to java

### PR DESCRIPTION
**Pull request by aman-agrawal**
Newly added java UTs are passing, Deleted corresponding groovy UTs

### CODE REVIEW

File: front50-core/src/test/java/com/netflix/spinnaker/front50/validator/HasCanonicalPluginIdValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2014 Netflix, Inc.
```
Comment (dbyron-sf) [inline front50-core/src/test/java/com/netflix/spinnaker/front50/validator/HasCanonicalPluginIdValidatorTest.java:2]:
```suggestion
 * Copyright 2019 Netflix, Inc.
```

File: front50-core/src/test/java/com/netflix/spinnaker/front50/validator/HasEmailValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    assertEquals(
+        numberOfErrors,
+        errors.getAllErrors().size(),
+        String.format("Expected %d error(s) for email: '%s'", numberOfErrors, email));
+  }
+
+  private static Stream<Object[]> emailProvider() {
+    return Stream.of(
+        new Object[] {"null", 1, "an empty"},
```
Comment (dbyron-sf) [inline front50-core/src/test/java/com/netflix/spinnaker/front50/validator/HasEmailValidatorTest.java:52]: Does
```suggestion
        new Object[] {null, 1, "an empty"},
```
work, and mean the special case for null above can disappear?

File: front50-core/src/test/java/com/netflix/spinnaker/front50/validator/HasNameValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    assertEquals(
+        numberOfErrors,
+        errors.getAllErrors().size(),
+        String.format("Expected %d error(s) for name: '%s'", numberOfErrors, name));
+  }
+
+  private static Stream<Object[]> nameProvider() {
+    return Stream.of(
+        new Object[] {"null", 1},
```
Comment (dbyron-sf) [inline front50-core/src/test/java/com/netflix/spinnaker/front50/validator/HasNameValidatorTest.java:52]:
```suggestion
        new Object[] {null, 1},
```
similar to above.

Comment (aman-agrawal) [reply front50-core/src/test/java/com/netflix/spinnaker/front50/validator/HasNameValidatorTest.java]: Done

File: front50-core/src/test/java/com/netflix/spinnaker/front50/validator/HasEmailValidatorTest.java (hunk 2 above)
Comment (aman-agrawal) [reply front50-core/src/test/java/com/netflix/spinnaker/front50/validator/HasEmailValidatorTest.java]: Done

File: front50-core/src/test/java/com/netflix/spinnaker/front50/validator/HasCanonicalPluginIdValidatorTest.java (hunk 1 above)
Comment (aman-agrawal) [reply front50-core/src/test/java/com/netflix/spinnaker/front50/validator/HasCanonicalPluginIdValidatorTest.java]: Done

**Review by dbyron-sf — APPROVED** [review]

**Pull request merged**